### DISCUSSION TITLE
Set INLINE_RUNTIME_CHUNK=false in Next.js

### DISCUSSION THREAD

**Discussion by gm-rebricker** [vercel/next.js#31781] [Help]
Hello,

I try to configure the csp header for my Next.JS project.

To do this I need to remove the need of  'unsafe-eval' in the script-src.
A google search showed me that in plain react the enviroment-variable "INLINE_RUNTIME_CHUNK=false" prevents the generation of inline-scripts.

Is this possible to configure in Next.JS too?

I set the env-variable but it didn't change anything.

Example error message:

`EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'nonce-iyTL2JJ0WIK1HRnss99JtI9U8IvQHppxKQt6QaUgMKg=' 'strict-dynamic' 'self' ".

    at new Function (<anonymous>)
    at 509-87c87f05b61b37b8a3ef.js:1
    at Object.useMemo (framework-c93ed74a065331c4bd75.js:1)
    at Object.t.useMemo (framework-c93ed74a065331c4bd75.js:1)
    at t.R (509-87c87f05b61b37b8a3ef.js:1)
    at ro (framework-c93ed74a065331c4bd75.js:1)
    at Hu (framework-c93ed74a065331c4bd75.js:1)
    at Pi (framework-c93ed74a065331c4bd75.js:1)
    at xi (framework-c93ed74a065331c4bd75.js:1)
    at _i (framework-c93ed74a065331c4bd75.js:1)`

**Comment by leandro-ortiz-electric** [comment]
I guess it works only with apps created using `create-react-app` tool.
For NextJS the solution is more complex. Here are some examples:
https://stackoverflow.com/questions/65551212/using-csp-in-nextjs-nginx-and-material-uissr
https://guydumais.digital/blog/how-to-deploy-a-strict-content-security-policy-csp-with-next-js/
https://reesmorris.co.uk/blog/implementing-proper-csp-nextjs-styled-components

**Comment by leerob** [comment]
Hey folks, wanted to swing back here with an update. After digging through many different issues and discussions, I've [**made a new page in the documentation**](https://nextjs.org/docs/app/building-your-application/configuring/content-security-policy) ([PR](https://github.com/vercel/next.js/pull/54601)) specifically for Content Security Policy and nonces. This docs page: 

- Explains how to generate a `nonce` with Middleware
- Shows how to consume the `nonce` in a route with `headers()`
- Shows a complete CSP without needing to use any `unsafe`
- Shows how to ignore the `nonce` Middleware from running on prefetches / static assets

Further, we've patched some bugs and made improvements to `nonce` handling in Next.js that will be available in the latest `canary` version (for those of you time traveling from the future, upgrade to Next.js 13.5). We also updated the `with-strict-csp` example in the `examples/` folder, which is backlinked from the [new documentation page](https://nextjs.org/docs/app/building-your-application/configuring/content-security-policy).

Really hope this helps out, thank you all 🙏 I'll be closing this discussion out. To continue the discussion, [please go here](https://github.com/vercel/next.js/discussions/54907).
